embed/etcd.go

```diff
@@ -318,16 +318,25 @@ func (e *Etcd) serve() (err error) {
 		}(l)
 	}
 
-	// Start a client server goroutine for each listen address
-	ch := http.Handler(&cors.CORSHandler{
-		Handler: v2http.NewClientHandler(e.Server, e.Server.Cfg.ReqTimeout()),
-		Info:    e.cfg.CorsInfo,
-	})
+	v2h := func() http.Handler {
```

Comment:
Please nil this out if v2 is disabled, instead of having a handler that returns an error, and build a v3-only handler in `grpcHandlerFunc` in serve.go if it's nil like I suggested? The reasoning is that if the v2 endpoint is disabled, having a v2 handler only to respond with an error only adds extra overhead to request processing.